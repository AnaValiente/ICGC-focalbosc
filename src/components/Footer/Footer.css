.containerFooter{
	width: 100%;
	max-height: 3em;
	display: flex;
	flex-direction: row;
	align-items: center;
	padding: 0.5em;
	background-color:#FA481E;
	position: absolute;
	bottom: 0;
	z-index: 1000;
	justify-content: space-between;
}

.containerFooterMobile{
	width: 100%;
	max-height: 3em;
	display: flex;
	flex-direction: row;
	align-items: center;
	padding: 0.5em;
	background-color:#FA481E;
	position: absolute;
	bottom: 0;
	z-index: 1000;
	justify-content: center;
}

.myInvertedButton{
	box-shadow: none !important;
	background: transparent none !important;
	color: #fff !important;
	text-shadow: none!important;
}

.myInvertedButton:hover, .myInvertedButton:focus {
	background: #fff !important;
	box-shadow: 0 0 0 2px #fff inset!important;
	color: rgba(0,0,0,.8) !important;
}

.description{
	width:100%;
}

.containerRightMobile{
	align-self: flex-end;
}