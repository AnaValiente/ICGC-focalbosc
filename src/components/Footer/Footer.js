// @flow

import React, { useState }  from "react";
import {  Button, Divider, Form, Icon } from "semantic-ui-react";
import { ResizeComponent } from "@geostarters/react-components";

import ModalInfo from "../ModalInfo/ModalInfo";
import styles from "./Footer.module.css";
import {
	URLMAIL,
	URLTWITTER,
	URLFACEBOOK,
	URL_GITHUB,
	URL_PROTOTIPS
} from "../../constants";


const Footer = (props) => {

	const [currentUrl, setCurrentUrl] = useState(window.location.href);

	const isMobile = props.width <= 500;

	const openLink = (url, mode = "_blank") => window.open(url, mode);

	const getEncodedUrlApp = () => encodeURI(window.location.href);

	const getEmbedUrlApp = (url) => `<div style="height: 75vh; width: 100%; margin: auto;"><iframe width="100%" height="100%" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" src="${url}" ></iframe></div>`;

	const renderBTLink = () => (
		<ModalInfo modalSize="tiny" modalTrigger={<Button className={styles.myInvertedButton} onClick={() => setCurrentUrl(getEncodedUrlApp())}><Icon name='linkify' /> Enllaça</Button>}>
			<h3>Enllaça la vista</h3>
			<Divider />
			<Form inline="true" >
				<Form.Field >
					<label>Per enllaçar amb aquest mapa, copieu i enganxeu el següent text:</label>
					<input readOnly value={currentUrl} />
				</Form.Field>
				<Form.Field>
					<label>Per inserir aquest mapa al vostre web, copieu i enganxeu el següent text:</label>
					<textarea readOnly value={getEmbedUrlApp(currentUrl)} />
				</Form.Field>
			</Form>
		</ModalInfo>
	);

	const renderContainerLeft = () => (
		<div className={styles.containerLeft}>
			<Button className={styles.myInvertedButton} onClick={() => openLink(URL_PROTOTIPS)}>
				<Icon name='external alternate' /> +Prototips
			</Button>
			<Button  className={styles.myInvertedButton} onClick={() => openLink(URL_GITHUB)}>
				<Icon name='github' /> Github
			</Button>
			{renderBTLink()}
		</div>
	);

	const renderContainerRight = () => (
		<div className={styles.containerRight}>
			<Button onClick={()=>openLink(`${URLMAIL}${getEncodedUrlApp()}`, "_self")} circular inverted icon='mail' />
			<Button onClick={()=>openLink(`${URLTWITTER}${getEncodedUrlApp()}`)} circular inverted icon='twitter' />
			<Button onClick={()=>openLink(`${URLFACEBOOK}${getEncodedUrlApp()}`)} circular inverted icon='facebook' />
		</div>
	);

	return (
		<div className={props.width <= 500 ? styles.containerFooterMobile : styles.containerFooter}>
			{!isMobile && renderContainerLeft()}
			{renderContainerRight() }
		</div>
	);

};

Footer.DefaultProperties = {
	width: 800
};

// eslint-disable-next-line new-cap
export default React.memo(ResizeComponent(Footer));
