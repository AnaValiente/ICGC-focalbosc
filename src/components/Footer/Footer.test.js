//@flow

import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Footer from "./Footer";
import {  Button } from "semantic-ui-react";

describe("Footer component", () => {

	it("renders correct properties ", ()=> {

		const mockProps = {
			width: 1000
		};

		const component = shallow(<Footer {...mockProps}></Footer>);
		expect(toJSON(component)).toMatchSnapshot();

	});

	it("renders correctly for mobile", ()=> {

		const mockProps = {
			width: 400
		};

		const component = mount(<Footer {...mockProps}></Footer>);
		expect(component.find(".containerLeft")).toHaveLength(0);
		expect(component.find(".containerRight")).toHaveLength(1);
		expect(component.find(Button)).toHaveLength(3);

		component.unmount();

	});


});
