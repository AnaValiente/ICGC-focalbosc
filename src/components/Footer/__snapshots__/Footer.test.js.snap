// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correct properties  1`] = `
<Footer
  width={1000}
/>
`;
